google-cloud-firestore/src/main/java/com/google/cloud/firestore/FirestoreImpl.java

```diff
@@ -279,19 +330,27 @@ public void run() {
 
           private void maybeRetry() {
             if (attemptsRemaining > 0) {
-              runTransaction(
+              span.addAnnotation("retrying");
+              runTransactionAttempt(
                   transactionCallback,
                   resultFuture,
                   new TransactionOptions(
-                      attemptsRemaining, options.getExecutor(), transaction.getTransactionId()));
+                      attemptsRemaining, options.getExecutor(), transaction.getTransactionId()),
+                  span);
             } else {
+              span.setStatus(TOO_MANY_RETRIES_STATUS);
+              span.end();
```

Comment:
Don't need to call end here because you call in the rejectTransaction(). Or if you want to have this specific status and end here probably better to pass a boolean to rejectTransaction to end or not the Span.

The call of the setStatus will be ignored if happens after end, also the second end is a no-op. But better to not do that in my opinion to make the code look better.

Reply:
I removed the `span.end()` here. I left the status, since I think it's useful to preserve "too many retries" information.
Status shouldn't be double set by rejectTransaction since FirestoreException is not an instanceof ApiException.